modules/isofs/zso.c

```diff
@@ -0,0 +1,103 @@
+#include "lz4.h"
+#include "zso.h"
+
+// block offset cache, reduces IO access
+u32 ciso_idx_cache[CISO_IDX_MAX_ENTRIES];
+int ciso_idx_start_block = -1;
+
+// header data that we need for the reader
+u64 ciso_uncompressed_size;
+u32 ciso_align;
+u32 ciso_total_block;
+
+// block buffers
+u8 ciso_dec_buf[2048] __attribute__((aligned(64)));
+u8 ciso_com_buf[3072] __attribute__((aligned(64)));
```

Comment:
5KiB

Reply:
This is the minimum you can have, maybe can be lowered to 4KB.